### PR TITLE
Add DNS zones data source

### PR DESCRIPTION
**Pull request by X4mp**
### Description

This PR adds the DNS zones data source.

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAnxCloudDnsZonesDataSource'                                                                                                                                              
==> Checking that code complies with gofmt requirements...                                                                                        
TF_ACC=1 go test $(go list ./... | grep -v 'vendor') -v -run=TestAccAnxCloudDnsZonesDataSource -timeout 120m -parallel=4                          
?       github.com/anexia-it/terraform-provider-anxcloud        [no test files]                                                                   
=== RUN   TestAccAnxCloudDnsZonesDataSource                                                                                                       
--- PASS: TestAccAnxCloudDnsZonesDataSource (3.32s)                                                                                               
PASS                                                                                                                                              
ok      github.com/anexia-it/terraform-provider-anxcloud/anxcloud       3.333s    

$ make testacc TESTARGS='-run=TestFlattenDnsZones'  
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... | grep -v 'vendor') -v -run=TestFlattenDnsZones -timeout 120m -parallel=4
?       github.com/anexia-it/terraform-provider-anxcloud        [no test files]
=== RUN   TestFlattenDnsZones
--- PASS: TestFlattenDnsZones (0.00s)
PASS
ok      github.com/anexia-it/terraform-provider-anxcloud/anxcloud       0.012s


$ make testacc TESTARGS='-run=TestFlattenDnsRecords'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... | grep -v 'vendor') -v -run=TestFlattenDnsRecords -timeout 120m -parallel=4
?       github.com/anexia-it/terraform-provider-anxcloud        [no test files]
=== RUN   TestFlattenDnsRecords
--- PASS: TestFlattenDnsRecords (0.00s)
PASS
ok      github.com/anexia-it/terraform-provider-anxcloud/anxcloud       0.015s

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/anexia-it/terraform-provider-anxcloud/blob/master/CHANGELOG.md):

```release-note
-  Added DNS zones data source
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: anxcloud/schema_dns_zone.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+					},
+					"is_master": {
+						Type:        schema.TypeBool,
+						Computed:    true,
+						Description: "Indicator if the zone is a master zone",
+					},
+					"dns_sec_mode": {
+						Type:        schema.TypeString,
+						Computed:    true,
+						Description: "DNSSec mode for master zones",
```
Comment (LittleFox94) [inline anxcloud/schema_dns_zone.go:26]: please more docs

File: anxcloud/schema_dns_zone.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+					},
+					"expire": {
+						Type:        schema.TypeInt,
+						Computed:    true,
+						Description: "Expiration value",
+					},
+					"ttl": {
+						Type:        schema.TypeInt,
+						Computed:    true,
+						Description: "Retry value",
```
Comment (LittleFox94) [inline anxcloud/schema_dns_zone.go:51]: TTL, not Retry

File: anxcloud/schema_dns_zone.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+									Computed:    true,
+									Description: "DNS server alias",
+								},
+							},
+						},
+					},
+					"is_editable": {
+						Type:        schema.TypeBool,
+						Computed:    true,
+						Description: "Idicator if zone is editable",
```
Comment (LittleFox94) [inline anxcloud/schema_dns_zone.go:86]: Indicator*

File: anxcloud/schema_dns_zone.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+					},
+					"validation_level": {
+						Type:        schema.TypeInt,
+						Computed:    true,
+						Description: "Zone validation level",
```
Comment (LittleFox94) [inline anxcloud/schema_dns_zone.go:91]: more docs pls

File: anxcloud/schema_dns_zone.go [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+					},
+					"deployment_level": {
+						Type:        schema.TypeInt,
+						Computed:    true,
+						Description: "Zone deployment level",
```
Comment (LittleFox94) [inline anxcloud/schema_dns_zone.go:96]: more docs pls

File: anxcloud/struct_dns_zones.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import "github.com/anexia-it/go-anxcloud/pkg/clouddns/zone"
+
+func flattenDnsZones(dnsZones []zone.Zone) []interface{} {
+	zones := make([]interface{}, 0, len(dnsZones))
+	if len(dnsZones) < 1 {
+		return zones
+	}
+
+	for _, zone := range dnsZones {
+		dnsServers := []interface{}{}
```
Comment (LittleFox94) [inline anxcloud/struct_dns_zones.go:12]: `make(...)`

File: anxcloud/struct_dns_zones.go [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		for _, dnsServer := range zone.DNSServers {
+			d := map[string]interface{}{}
+			d["server"] = dnsServer.Server
```
Comment (LittleFox94) [inline anxcloud/struct_dns_zones.go:15]:
```go
{
  "server": ...,
  "alias": ...,
}
```

File: anxcloud/struct_dns_zones_test.go [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+							},
+						},
+					},
+					DeploymentLevel: 99,
+					ValidationLevel: 99,
+					IsEditable:      false,
+				},
+			},
+			[]interface{}{
+				map[string]interface{}{
```
Comment (LittleFox94) [inline anxcloud/struct_dns_zones_test.go:64]: hm when we are using this type outside of a single function, we should really name it ..

File: anxcloud/struct_dns_zones_test.go [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+					"validation_level":   100,
+					"is_editable":        true,
+					"dns_servers": []interface{}{
+						map[string]interface{}{
+							"server": "nameserver-1",
+							"alias":  "ns1",
+						},
+					},
+				},
+				map[string]interface{}{
```
Comment (LittleFox94) [inline anxcloud/struct_dns_zones_test.go:85]: same as above

File: anxcloud/struct_dns_zones_test.go [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+					"retry":              300,
+					"expire":             3600,
+					"ttl":                60,
+					"notify_allowed_ips": []string{"127.0.0.1", "192.168.0.1"},
+					"master_nameserver":  "8.8.8.8",
+					"deployment_level":   99,
+					"validation_level":   99,
+					"is_editable":        false,
+					"dns_servers": []interface{}{
+						map[string]interface{}{
```
Comment (LittleFox94) [inline anxcloud/struct_dns_zones_test.go:100]: another type in need of naming I think

**Review by LittleFox94 — CHANGES_REQUESTED** [review]

File: anxcloud/struct_dns_zones_test.go (hunk 10 above)
Comment (LittleFox94) [reply anxcloud/struct_dns_zones_test.go:100]: scrap that, consistency is more important than this one

File: anxcloud/struct_dns_zones_test.go (hunk 8 above)
Comment (LittleFox94) [reply anxcloud/struct_dns_zones_test.go:64]: scrap that, consistency is more important than this one

Comment (X4mp) [reply anxcloud/struct_dns_zones_test.go:64]: Due to consistnecy I would like to keep it this way.

File: anxcloud/struct_dns_zones.go [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+func flattenDnsZones(dnsZones []zone.Zone) []interface{} {
+	zones := make([]interface{}, 0, len(dnsZones))
+	if len(dnsZones) < 1 {
+		return zones
+	}
+
+	for _, zone := range dnsZones {
+		dnsServers := make([]interface{}, 0, len(zone.DNSServers))
+		for _, dnsServer := range zone.DNSServers {
```
Comment (LittleFox94) [inline anxcloud/struct_dns_zones.go:13]: come on, empty lines come at no cost at all

(ok, except one byte)

Comment (LittleFox94) [reply anxcloud/struct_dns_zones.go:13]: (between dnsServers := ... and the for loop)

**Pull request merged**